state/modelgeneration.go

```diff
@@ -284,7 +260,8 @@ func (g *Generation) allowMakeCurrent(allowEmpty bool) (bool, error) {
 	}
 	if !ok {
 		if allowEmpty {
-			return false, errors.New(fmt.Sprintf("cannot cancel generation, there are units behind a generation: %s", strings.Join(values, ", ")))
+			return false, errors.Errorf(
```

Comment:
If we make this a typed error, then I think the 'advance' call can just say "if errors.Cause(err) == GenerationNotComplete"
rather than doing a pre-check.
Certainly we have to do all the same work in complete(false) that we would do if we wanted "CanAutocomplete" right?

Reply:
Yes; I have done this.